### PR TITLE
Show page views in action=info

### PR DESCRIPTION
**Pull request by AgentIsai**

### CODE REVIEW

File: includes/MatomoAnalyticsWiki.php [hunk 1]
```diff
@@ -131,4 +138,15 @@ public function getVisitDurations() {
 	public function getVisitDaysPassed() {
 		return $this->getData( 'VisitorInterest.getNumberOfVisitsByDaysSinceLast' );
 	}
+
+	// Visits by social network
```
Comment (Universal-Omega) [inline includes/MatomoAnalyticsWiki.php:142]: What???

File: includes/MatomoAnalyticsHooks.php [hunk 2]
```diff
@@ -91,4 +93,24 @@ public static function matomoScript( $skin, &$text ) {
...
 		return true;
 	}
+
+		/**
+	 * Display total pageviews in the last 30 days and show a graph with details when clicked.
+	 * @param IContextSource $context
+	 * @param array &$pageInfo
+	 */
+	public function onInfoAction( $context, &$pageInfo ) {
+		$mA = new MatomoAnalyticsWiki( $context->getConfig()->get( 'DBname' ) );
```
Comment (Universal-Omega) [inline includes/MatomoAnalyticsHooks.php:103]: MainConfigNames

File: includes/MatomoAnalyticsHooks.php [hunk 3]
```diff
@@ -91,4 +93,24 @@ public static function matomoScript( $skin, &$text ) {
...
+
+		$title = $context->getTitle();
+		$data = $mA->getPageViews( $title );
+		$total = array_sum( $data );
+
+		$lang = $context->getLanguage();
+		$formatted = $lang->formatNum( $total );
```
Comment (BlankEclair) [inline includes/MatomoAnalyticsHooks.php:110]: Can be simplified with [`Message::numParams()`](https://doc.wikimedia.org/mediawiki-core/REL1_42/php/classMediaWiki_1_1Message_1_1Message.html#a49ede4e918e1f5834396ca41cae58c9d)

File: extension.json [hunk 4]
```diff
@@ -24,33 +24,45 @@
...
+		"InfoAction": {
+			"handler": "Main"
+		},
+		"SkinAfterBottomScripts": {
+			"handler": "Main"
+		}
+	},
+	"HookHandlers": {
+		"CreateWiki": {
+			"class": "Miraheze\\MatomoAnalytics\\Hooks\\Handlers\\CreateWiki"
```
Comment (Universal-Omega) [inline extension.json:58]: Use HookHandlers not Hooks\Handlers

**Review by Universal-Omega — CHANGES_REQUESTED** [review]

File: includes/MatomoAnalyticsWiki.php [hunk 5]
```diff
@@ -15,22 +17,29 @@ private function getData(
 		string $period = 'range',
 		string $jsonLabel = 'label',
 		string $jsonData = 'nb_visits',
-		bool $flat = false
+		bool $flat = false,
+		string $title = null
```
[the remark this answers is not in the document]
Comment (Universal-Omega) [reply includes/MatomoAnalyticsWiki.php:21]: Technically this doesn't need `?`, however that does change behavior in PHP 8.4 where is does require to be explicitly nullable and = null doesn't make it implicit anymore (or at least deprecated) so yeah it definitely should be now for future support.

**Review by BlankEclair — APPROVED** [review]

File: .github/workflows/mediawiki-tests.yml [hunk 6]
```diff
@@ -12,17 +12,18 @@ jobs:
     strategy:
       matrix:
         include:
-          # Latest MediaWiki stable - PHP 8.1
+          # Latest MediaWiki stable - PHP 8.2
```
Comment (Universal-Omega) [inline .github/workflows/mediawiki-tests.yml:15]: This needs to be the same on all extensions so we should not be changing mediawiki-tests in extension PRs. Also 8.1 should be kept.

File: .github/workflows/mediawiki-tests.yml [hunk 7]
```diff
@@ -12,17 +12,18 @@ jobs:
...
           - mw: 'REL1_42'
-            php: 8.1
-            php-docker: 81
+            php: 8.2
+            php-docker: 82
             composer-test: true
             experimental: false
+            keepdependencies: true
```
Comment (Universal-Omega) [inline .github/workflows/mediawiki-tests.yml:21]: This isn't a valid option

File: extension.json [hunk 8]
```diff
@@ -6,7 +6,7 @@
 		"Universal Omega"
 	],
 	"url": "https://github.com/miraheze/MatomoAnalytics",
-	"version": "1.1.3",
+	"version": "1.2.0",
```
Comment (Universal-Omega) [inline extension.json:9]: CHANGELOG entry?

File: i18n/en.json [hunk 9]
```diff
@@ -2,18 +2,21 @@
...
 			"John Lewis",
-			"Universal Omega"
+			"Universal Omega",
+			"Agent Isai"
 		]
 	},
 	"action-noanalytics": "not be tracked via Matomo",
-	"analytics": "MatomoAnalytics",
+	"analytics": "Wiki analytics",
+	"matomoanalytics-count": "$1",
```
Comment (Universal-Omega) [inline i18n/en.json:11]: qqq entries?

File: maintenance/cleanupMatomos.php [hunk 10]
```diff
@@ -49,7 +55,8 @@ public function execute() {
 				}
 
 				$this->output( "Remove matomo id from {$DBname}\n" );
-				MatomoAnalytics::deleteSite( $DBname );
+				$mA = new MatomoAnalytics;
+				$mA->deleteSite( $DBname );
```
Comment (Universal-Omega) [inline maintenance/cleanupMatomos.php:59]: Static function we shouldn't be using new MatomoAnalytics

File: maintenance/modifyMatomo.php [hunk 11]
```diff
@@ -19,10 +24,13 @@ public function __construct() {
 
 	public function execute() {
 		$DBname = $this->getConfig()->get( 'DBname' );
+		$mA = new MatomoAnalytics;
 
-		$this->getOption( 'remove', false ) ?
-			MatomoAnalytics::deleteSite( $DBname ) :
-			MatomoAnalytics::addSite( $DBname );
+		if ( $this->getOption( 'remove', false ) ) {
+			$mA->deleteSite( $DBname );
```
Comment (Universal-Omega) [inline maintenance/modifyMatomo.php:30]: Static function we shouldn't be using new MatomoAnalytics

File: maintenance/modifyMatomo.php [hunk 12]
```diff
@@ -19,10 +24,13 @@ public function __construct() {
...
+		} else {
+			$mA->addSite( $DBname );
```
Comment (Universal-Omega) [inline maintenance/modifyMatomo.php:32]: Static function we shouldn't be using new MatomoAnalytics

File: maintenance/addMissingMatomos.php [hunk 13]
```diff
@@ -42,8 +48,10 @@ public function execute() {
 			);
 
 			if ( !isset( $id ) || !$id ) {
-				$this->output( "Add matomo id to {$DBname}\n" );
-				MatomoAnalytics::addSite( $DBname );
+				$this->output( "Adding matomo id to {$DBname}\n" );
+				$mA = new MatomoAnalytics;
+				$mA->addSite( $DBname );
```
Comment (Universal-Omega) [inline maintenance/addMissingMatomos.php:53]: Static function we shouldn't be using new MatomoAnalytics

File: includes/HookHandlers/CreateWiki.php [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+	CreateWikiRenameHook
+{
+	public function onCreateWikiCreation( string $dbname, bool $private ): void {
+		$mA = new MatomoAnalytics;
+		$mA->addSite( $dbname );
+	}
+
+	public function onCreateWikiDeletion( DBConnRef $cwdb, string $dbname ): void {
+		$mA = new MatomoAnalytics;
+		$mA->deleteSite( $dbname );
```
Comment (Universal-Omega) [inline includes/HookHandlers/CreateWiki.php:23]: Static function we shouldn't be using new MatomoAnalytics

File: includes/HookHandlers/CreateWiki.php [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+	}
+
+	public function onCreateWikiRename( DBConnRef $cwdb, string $old, string $new ): void {
+		$mA = new MatomoAnalytics;
+		$mA->renameSite( $old, $new );
```
Comment (Universal-Omega) [inline includes/HookHandlers/CreateWiki.php:28]: Static function we shouldn't be using new MatomoAnalytics

File: includes/HookHandlers/CreateWiki.php [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+use Wikimedia\Rdbms\DBConnRef;
+
+class CreateWiki implements
+	CreateWikiCreationHook,
+	CreateWikiDeletionHook,
+	CreateWikiRenameHook
+{
+	public function onCreateWikiCreation( string $dbname, bool $private ): void {
+		$mA = new MatomoAnalytics;
+		$mA->addSite( $dbname );
```
Comment (Universal-Omega) [inline includes/HookHandlers/CreateWiki.php:18]: Static function we shouldn't be using new MatomoAnalytics

**Pull request merged**